Userland/Libraries/LibWeb/Bindings/LocationObject.cpp

```diff
@@ -222,6 +223,17 @@ JS_DEFINE_NATIVE_FUNCTION(LocationObject::port_getter)
     return JS::js_string(vm, String::number(*location_object->url().port()));
 }
 
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-location-origin
+JS_DEFINE_NATIVE_FUNCTION(LocationObject::origin_getter)
+{
+    auto* location_object = TRY(typed_this_value(vm));
+
+    // FIXME: 1. If this's relevant Document is non-null and its origin is not same origin-domain with the entry settings object's origin, then throw a "SecurityError" DOMException.
```

Comment:
this should be implementable as something like

```c++
auto* document = location_object->relevant_document();
if (document && !document->origin().is_same_origin_domain(Bindings::host_defined_environment_settings_object(*vm().current_realm()).origin()))
    return throw_completion(WebIDL::SecurityError::create(*vm.current_realm(), "Some descriptive error message"sv));
```

:thinking: but I suppose that every function in this file has the same fixme, so maybe that's ok to leave for later.

probably worth a helper called `EnvironmentSettingsObject& entry_settings_object(JS::VM&)` as well.

Reply:
Yes, I also thought it would be good for the same reason (and also because I didn't know how to do it). But anyway, I'd better not touch it so I don't have to change the PR scope, probably this would be better done in another PR.